native/exe/ServiceControlLaunch/ServiceControlLaunch.cpp

```diff
@@ -126,44 +132,75 @@ int APIENTRY WinMain(HINSTANCE hInstance,
 	DWORD dwRead = 0;
 	DWORD hsize = sizeof(dwRead);
 	hsize = sizeof(currVer);
-	if (RegOpenKeyEx(rootKey, "Software\\JavaSoft\\Java Runtime Environment", 0, KEY_QUERY_VALUE, &myKey) != ERROR_SUCCESS)
+	HMODULE exeMod = GetModuleHandle(NULL);

```

Comment:
If you want a good learning project...what really SHOULD have been done here when I wrote it was to refactor this into a common source file that's used by all 3 launchers which can invoke the JVM (and other reused stuff like sysOutPrint). That avoids code duplication and will make them all consistent.

Reply:
I did notice there's a lot of common code.  :)
I'll see if I can figure out how to do that for a future change